actors/multisig/tests/multisig_actor_test.rs

```diff
@@ -1090,46 +1090,481 @@ fn test_signer_swap() {
+        let bob = Address::new_id(102);
+        let chuck = Address::new_id(103);
+        let signers = vec![anne, bob, chuck];
+        let mut rt = construct_runtime(msig);
+        let send_value = TokenAmount::from(10u8);
+        let h = util::ActorHarness::new();
+        rt.set_balance(send_value.clone());
+        rt.set_received(TokenAmount::zero());
+        h.construct_and_verify(&mut rt, 3, 0, 0, signers);
+
+        let fake_params = RawBytes::from(vec![1, 2, 3, 4]);
+        let fake_method = 42;
+        rt.set_caller(*ACCOUNT_ACTOR_CODE_ID, anne);
+        let proposal_hash =
+            h.propose_ok(&mut rt, chuck, send_value.clone(), fake_method, fake_params.clone());
+
+        // bob approves tx
+        rt.set_caller(*ACCOUNT_ACTOR_CODE_ID, bob);
+        h.approve_ok(&mut rt, TxnID(0), proposal_hash);
+
+        // reduce threshold so tx is already approved
+        rt.set_caller(*MULTISIG_ACTOR_CODE_ID, msig);
+        let new_threshold = 2;
+        h.change_num_approvals_threshold(&mut rt, new_threshold).unwrap();
+
+        // duplicate approval executes tx because the msig is across the threshold
+        rt.expect_send(
+            chuck,
+            fake_method,
+            fake_params,
+            send_value,
+            RawBytes::default(),
+            ExitCode::OK,
+        );
+        rt.set_caller(*ACCOUNT_ACTOR_CODE_ID, bob);
+        h.approve_ok(&mut rt, TxnID(0), proposal_hash);
+        h.assert_transactions(&rt, vec![]);
+    }
+
+    #[test]
+    fn approve_tx_if_num_approvers_has_already_crossed_threshold_but_non_signatory_cannot_approve_tx(
+    ) {
+        let msig = Address::new_id(100);
+        let anne = Address::new_id(101);
+        let bob = Address::new_id(102);
+        let chuck = Address::new_id(103);
+        let signers = vec![anne, bob];
+        let mut rt = construct_runtime(msig);
+        let send_value = TokenAmount::from(10u8);
+        let h = util::ActorHarness::new();
+        rt.set_balance(send_value.clone());
+        rt.set_received(TokenAmount::zero());
+        h.construct_and_verify(&mut rt, 2, 0, 0, signers);
+
+        let fake_params = RawBytes::from(vec![1, 2, 3, 4]);
+        let fake_method = 42;
+        rt.set_caller(*ACCOUNT_ACTOR_CODE_ID, anne);
+        let proposal_hash =
+            h.propose_ok(&mut rt, chuck, send_value.clone(), fake_method, fake_params.clone());
+
+        // reduce threshold so tx is already approved
+        rt.set_caller(*MULTISIG_ACTOR_CODE_ID, msig);
+        let new_threshold = 1;
+        h.change_num_approvals_threshold(&mut rt, new_threshold).unwrap();
+
+        // non-signer alice cannot approve the tx
+        let alice = Address::new_id(104);
+        rt.set_caller(*ACCOUNT_ACTOR_CODE_ID, alice);
+        expect_abort(ExitCode::USR_FORBIDDEN, h.approve(&mut rt, TxnID(0), proposal_hash));
+        rt.reset();
+
+        // bob attempts to approve and it goes through immediately
```

Comment:
This part is a bit weird, because Bob's approval would have made it go through even without the change in the threshold. But I think it's fine. 

Could change it to Anne self-approving to make it a bit more clear?